src/node/channels.h

```diff
@@ -476,18 +476,25 @@ namespace ccf
     {
       std::lock_guard<SpinLock> guard(lock);
       auto search = channels.find(peer_id);
-      if (search != channels.end() && !search->second->is_outgoing())
+      if (search == channels.end())
+      {
+        LOG_DEBUG_FMT(
+          "Creating new outbound channel to {} ({}:{})",
+          peer_id,
+          hostname,
+          service);
+        auto channel = std::make_shared<Channel>(
+          writer_factory, network_kp, self, peer_id, hostname, service);
+        channels.emplace_hint(search, peer_id, std::move(channel));
+      }
+      else if (!search->second->is_outgoing())
       {
         // Channel with peer already exists but is incoming. Create host
         // outgoing connection.
+        LOG_DEBUG_FMT("Setting existing channel to {} as outgoing", peer_id);
         search->second->set_outgoing(hostname, service);
         return;
       }
```

Comment:
@achamayou + @jumaffre: Highlighting this related bugfix that this PR exposed. The `try_emplace` was unafe; if the channel already exist then we briefly construct a temporary `Channel`, and then immediately call its destructor which sends a message to the host closing the outgoing connection.

I don't fully grok why this is happening so often with these PR changes - something something candidate-to-candidate transitions are more common something something? But hopefully this clears up the CI instability on this PR, so I'm rolling the fix in here.